### DISCUSSION TITLE
Help with syntax -> using persists with TypeScript (Converting Circular structure...)

### DISCUSSION THREAD

**Discussion by Ettur** [pmndrs/zustand#867] [Q&A]
EDIT: I worked it out that with partialize I can choose which key: values to save so I only took token and it works!

EDIT: I think I figured out that problem is serializing set functions, I dont even really need to store them. I only really need token to be stored, is there a way I can select which store properties will be saved ?

I tried to implement persist on my store based on the example given but alas, I keep getting errors. Would someone knowledgeable edit my code to show me how to implement it ?

This is my store,

```
import { createRef, MutableRefObject } from "react";
import create, { GetState, SetState } from "zustand";
import { persist, StoreApiWithPersist, StoreApiWithSubscribeWithSelector, subscribeWithSelector  } from 'zustand/middleware'

interface View {
    current: string,
    next: string
}

interface AppState {
    navRef: MutableRefObject<HTMLDivElement | null> ;
    animRef: MutableRefObject<HTMLDivElement | null> ;
    view: View;
    token: string;
    setView: (view: View) => void;
    setToken: (token: string) => void;
}


export const useStore = create(
    subscribeWithSelector(
      persist<
        AppState,
        SetState<AppState>,
        GetState<AppState>,
        StoreApiWithSubscribeWithSelector<AppState> &
          StoreApiWithPersist<AppState>
      >(
        (set, get) => ({
            navRef: createRef(),
            animRef: createRef(),
            view: {current: 'landing', next: 'none'},
            token: '',
            setView: (view: View) => { set ({view: view})},
            setToken: (token: string) => { set ({token: token})}
        }),
        { name: 'name', getStorage: ()=> localStorage }
      )
    )
  )
```

Error I get is 

```
TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'HTMLDivElement'
    |     property '__reactFiber$28pflfz5u2h' -> object with constructor 'FiberNode'
    --- property 'stateNode' closes the circle
```

points to this line 
` setView: (view: View) => { set ({view: view})},`

**Comment by abhik-99** [comment]
Getting a similar problem. Any solutions to converting Circular structure (in my case its a provider object which is used for crypto wallet login)?

**Reply by hussain7college** [reply]
use this: https://docs.pmnd.rs/zustand/integrations/persisting-store-data#partialize

**Comment by hussain7college** [comment]
I think this solved the issue for me: https://docs.pmnd.rs/zustand/integrations/persisting-store-data#partialize

**Reply by chienzo123** [reply]
It good, tkss
